src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs

```diff
@@ -1080,5 +1086,82 @@ internal static bool IsShellInCommandLineMode(System.IServiceProvider servicePro
 
             return ((bool)isInCommandLineModeAsObject);
         }
+
+        /// <summary>
+        /// Builds a set of Inline elements from the specified message, without embedded hyperlinks.
+        /// </summary>
+        /// <param name="message">The message to process.</param>
+        /// <returns>A collection of Inline elements that represent the specified message.</returns>
+        internal static List<Inline> GetInlinesForErrorMessage(string message)
+        {
+            return GetInlinesForErrorMessage(message, null, null);
+        }
+
+        /// <summary>
+        /// Builds a set of Inline elements from the specified message, optionally with embedded hyperlinks.
+        /// </summary>
+        /// <param name="message">The message to process.</param>
+        /// <param name="createHyperlinks">True to generate embedded hyperlinks; otherwise, false.</param>
+        /// <param name="data">A data object to include in each hyperlink's Tag property.</param>
+        /// <param name="clickHandler">A delegate for the Hyperlink.Click event.</param>
+        /// <returns>A collection of Inline elements that represent the specified message.</returns>
+        internal static List<Inline> GetInlinesForErrorMessage(string message, object data, RoutedEventHandler clickHandler)
```

Comment:
Unless you foresee a case where a caller would pass something other than the index, IMO the parameter `object data` should be `int index`. From the point of view of the caller, that's what they're passing. The fact that internally, this message makes use of this data by storing it in an object-valued tag is of no concern to the caller.

Reply:
I made it an object because the consumer of it will be the caller's click handler, so this way they can stash whatever they need in there. (The Tuple is actually what's stored in the Tag.)
I'm not married to it though...

Reply:
Changed